nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -561,161 +559,201 @@ impl BundleCollection<'_> {
         .await?;
 
         if let Ok(all_sleds) = self
-            .collector
             .datastore
             .sled_list_all_batched(&self.opctx, SledFilter::InService)
             .await
         {
             report.listed_in_service_sleds = true;
 
-            // NOTE: This could be, and probably should be, done concurrently.
-            for sled in &all_sleds {
-                info!(&log, "Collecting bundle info from sled"; "sled" => %sled.id());
-                let sled_path = dir
-                    .path()
-                    .join("rack")
-                    .join(sled.rack_id.to_string())
-                    .join("sled")
-                    .join(sled.id().to_string());
-                tokio::fs::create_dir_all(&sled_path).await?;
-                tokio::fs::write(
-                    sled_path.join("sled.txt"),
-                    format!("{sled:?}"),
-                )
-                .await?;
+            const MAX_CONCURRENT_SLED_REQUESTS: usize = 16;
+            let mut sleds_iter = all_sleds.into_iter().peekable();
+            let mut tasks = JoinSet::new();
 
-                if self.request.skip_sled_info {
-                    continue;
+            // While we have incoming work to send to tasks (sleds_iter)
+            // or a task operating on that data (tasks)...
+            while sleds_iter.peek().is_some() || !tasks.is_empty() {
```

Comment:
nit: You could probably swap this to use a semaphore but I am not pressed about it.